mitreattack/attackToExcel/stixToDf.py

```diff
@@ -383,67 +516,19 @@ def matricesToDf(src, domain):
-                        subtechniques_column.append(subtechniques[i]["name"])
-                    technique_bottom = len(techniques_column) + 1  # bottom of row range to merge
-                    if technique_top != technique_bottom:  # more than 1 sub-technique
-                        merge.append(CellRange(  # merge technique portion of cell group
-                            len(columns),
-                            len(columns),
-                            technique_top,
-                            technique_bottom,
-                            data=technique["name"],
-                            format={  # format of the merged range
-                                "name": "supertechnique",
-                                "format": {
-                                    'valign': 'vcenter',
-                                    'text_wrap': 1,
-                                    'shrink': 1
-                                }
-                            }
-                        ))
-                else:  # no sub-techniques; add empty cell parallel to technique
-                    subtechniques_column.append("")
-            # end adding techniques and sub-techniques to column
-
-            matrix_grid.append(techniques_column)  # add technique column to grid
-            if len(list(filter(lambda x: x != "",
-                               subtechniques_column))) > 0:  # if there are sub-techniques for the tactic as well
-                matrix_grid.append(subtechniques_column)  # add sub-technique sub-column
-                columns.append("")  # add empty tactic header for the sub-column
-                merge.append(  # merge tactic column header with the sub-column header that was just appended
-                    CellRange(len(columns) - 1,
-                              len(columns),
-                              1,
-                              1,
-                              data=tactic["name"],
-                              format={  # tactic header formatting
-                                  "name": "tacticHeader",
-                                  "format": {
-                                      "bold": 1,
-                                      "border": 1,
-                                      "font_size": 14,
-                                      "align": "center",
-                                      "shrink": 1
-                                  }
-                              }
-                              ))
+            build_technique_and_sub_columns(src, techniques, columns, merge, matrix_grid, tactic['name'])
+
+            for platform in Matrix_Platforms_Lookup[domain]:
+                if platform.startswith('Google'):
+                    pass
+                a_techs = filter_platforms(techniques,
+                                           [platform] if platform not in Matrix_Platforms_Lookup
+                                           else Matrix_Platforms_Lookup[platform])
```

Comment:
This is ingenious but definitely needs to be documented with a comment around the `Matrix_Platforms_Lookup` definition. It's not otherwise clear that enterprise will look up the Cloud entry in the lookup.

Reply:
Yeah, I can definitely document this better. I've done so both here and at the definition.